microsoft/iot-cardboard-js
# Create flighted Visual rule viewer experience

### Summary of changes 🔍 
Added flighted capabilities to show Visual rules in viewer experience. With current mock data all should run smoothly. Since this functionality is still flighted and not forward compatible, I won't be able to update mock data until I remove flighting. Which will be done in next PR.

This PR is mostly duplicating and slightly modifying existing components that rendered the Elements list and Behavior section to be able to support Visual rules instead of Alerts/Status. Terminology for Alerts/Status is removed in the new components but is still around on the codebase, will target to remove that next PR. 

Since this PR contains a lot of duplicate components, the types and styles files haven't been updated to match the V2 name. Naming updates are targeted for next PR.

Here's a screenshot of all places that contain changes:
![image](https://user-images.githubusercontent.com/55852250/201239901-6f0f8857-924f-4840-a8c1-92011c229717.png)
- Left navbar, shows callout when color pills are hovered over. (Tooltip host cannot be added to a Cardboard list item).
- Small modal when hovering over a mesh now shows improved color pills.
- Right modal, shows updated color pills and callout shows up when row is hovered.
- All three spots support and correctly render several badges.

### Testing 🧪
- Contact me to get a modified Mock data file to be able to see the changes more clearly.
- Open browser dev tools and run this in console: `localStorage.setItem('cardboard.feature.visualRulesFeature',true);`
- Interact with objects to see their new behavior.
  - Change twin properties to be able to see how badges/colors get updated.

### Checklist ✔️
